zrml/prediction-markets/src/lib.rs

```diff
@@ -508,41 +509,44 @@ mod pallet {
                         Error::<T>::InsufficientFundsInMarketAccount,
                     );
 
-                    vec![(winning_currency_id, winning_balance)]
+                    vec![(winning_currency_id, winning_balance, winning_balance)]
                 }
                 Outcome::Scalar(value) => {
                     let long_currency_id = Asset::ScalarOutcome(market_id, ScalarPosition::Long);
                     let short_currency_id = Asset::ScalarOutcome(market_id, ScalarPosition::Short);
                     let long_balance = T::Shares::free_balance(long_currency_id, &sender);
                     let short_balance = T::Shares::free_balance(short_currency_id, &sender);
-                    let zero = BalanceOf::<T>::zero();
-                    let one = BalanceOf::<T>::one();
 
                     ensure!(
-                        long_balance >= zero || short_balance >= zero,
+                        long_balance >= BalanceOf::<T>::zero()
+                            || short_balance >= BalanceOf::<T>::zero(),
                         Error::<T>::NoWinningBalance
                     );
 
                     if let MarketType::Scalar((bound_low, bound_high)) = market.market_type {
-                        let calc_payouts =
-                            |final_value, low, high| -> (BalanceOf<T>, BalanceOf<T>) {
-                                if final_value <= low {
-                                    return (zero, one);
-                                }
-                                if final_value >= high {
-                                    return (one, zero);
-                                }
-
-                                let payout_long: u128 = (final_value - low) / (high - low);
-                                (
-                                    payout_long.saturated_into(),
-                                    one - payout_long.saturated_into(),
-                                )
-                            };
-
-                        let (long_payout, short_payout) =
+                        let calc_payouts = |final_value, low, high| -> (Percent, Percent) {
+                            if final_value <= low {
+                                return (Percent::zero(), Percent::one());
+                            }
+                            if final_value >= high {
+                                return (Percent::one(), Percent::zero());
+                            }
+
+                            let payout_long: Percent =
+                                Percent::from_rational(final_value - low, high - low);
+                            (
+                                payout_long,
+                                Percent::from_parts(
+                                    Percent::one().deconstruct() - payout_long.deconstruct(),
+                                ),
+                            )
```

Comment:
Suggestion: Use at least [Permill](https://substrate.dev/rustdocs/v3.0.0/sp_arithmetic/per_things/struct.Permill.html) for this, potentially even Perbill, because the precision for Percent is poor. In fact, the precision is 1/100 and the result is [floored to the next integer](https://crates.parity.io/sp_arithmetic/per_things/trait.PerThing.html#tymethod.from_rational), which leads to a potential inaccuracy of 0,99 (period) percent which in turn will be missing in the winning payouts.

Reply:
Good point